### PR TITLE
feature: add coverage feature

### PR DESCRIPTION
**Pull request by wallet77**

### CODE REVIEW

File: src/features/coverage.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    this.inspectorService.connect()
+    return this.inspectorService.post('Profiler.enable')
+  }
+
+  destroy () {
+    this.inspectorService.disconnect()
+  }
+
+  start (opts) {
+    this.method = !opts.method ? 'takePreciseCoverage' : opts.method
```
Comment (vmarchaud) [inline src/features/coverage.ts]: Im not sure why is the method set dynamically ? Is this required ?

File: src/features/coverage.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    return this.inspectorService.post('Profiler.startPreciseCoverage', { callCount: !!opts.callCount, detailed : !!opts.detailed })
+  }
+
+  async stop () {
+    const data = await this.getInfo()
+    return FileUtils.writeDumpFile(JSON.stringify(data), '.coverageprofile')
+  }
+
+  private getInfo () {
+    const self = this
```
Comment (vmarchaud) [inline src/features/coverage.ts]: If using an arrow function, i don't believe we need to access the context like this, could you confirm ?

File: src/features/coverage.ts (hunk 1 above)
Comment (wallet77) [reply src/features/coverage.ts]: Not required but can be useful to switch between takePreciseCoverage and getBestEffortCoverage
getBestEffortCoverage has a less impact on performance but can lost some data.

File: src/features/coverage.ts (hunk 2 above)
Comment (wallet77) [reply src/features/coverage.ts]: done

**Pull request merged**